### DISCUSSION TITLE
How does the export to URL work

### DISCUSSION THREAD

**Discussion by lambertjf** [jgraph/drawio#4161] [Q&A] [answered]
Hello, when exporting to URL, how is it encoded into the URL? Is it the XML converted to base64, or something along those lines? Is there a way I can do it myself if I have just the XML from a .drawio file?

I've tried using the convert tool in the drawio-tools to no avail. I input the XML from a diagram that I created and tried encoding it with all the options checked (base64, URL encode, and deflate) and then trying to create a URL from it in the same format that URL exporting does and I get an error going to that URL.

I tried the other way as well. I generated a URL from a diagram I created, took the encoded string off the end of the URL and tried to decode it. First URL decoding then base64/inflate, and I don't get proper XML. I'm guessing either I'm encoding/decoding this string wrong or when using the URL export method it doesn't use the XML at all or just doesn't use it entirely. Been going through a lot of different files in the project and haven't found an answer so far, some help would be great.

**Comment by alderg** [comment, marked as answer]
The #R stands for "raw", so that needs to be removed. The rest should work as expected:

<img width="880" alt="Screenshot 2024-02-01 at 12 10 23" src="https://github.com/jgraph/drawio/assets/921200/e7d00b13-6036-4330-afe7-83df65bae7a9">

**Reply by alderg** [reply]
In some cases you may have to click URL Decode and then Decode.

**Reply by lambertjf** [reply]
You are a hero, I was removing the # but leaving the R thinking it was part of the base64. Thanks so much
